Headers/AppKit/NSGridView.h

```diff
@@ -0,0 +1,212 @@
+/* Definition of class NSGridView
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 08-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#ifndef _NSGridView_h_GNUSTEP_GUI_INCLUDE
+#define _NSGridView_h_GNUSTEP_GUI_INCLUDE
+
+#import <AppKit/NSView.h>
+#import <AppKit/NSLayoutAnchor.h>
+
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_12, GS_API_LATEST)
+
+#if	defined(__cplusplus)
+extern "C" {
+#endif
+  
+enum {
+  NSGridCellPlacementInherited = 0,
+  NSGridCellPlacementNone,
+  NSGridCellPlacementLeading,
+  NSGridCellPlacementTop = NSGridCellPlacementLeading,    
+  NSGridCellPlacementTrailing,
+  NSGridCellPlacementBottom = NSGridCellPlacementTrailing,    
+  NSGridCellPlacementCenter,
+  NSGridCellPlacementFill    
+};
+typedef NSInteger NSGridCellPlacement;
+
+enum { 
+  NSGridRowAlignmentInherited = 0,
+  NSGridRowAlignmentNone,
+  NSGridRowAlignmentFirstBaseline,
+  NSGridRowAlignmentLastBaseline
+};
+typedef NSInteger NSGridRowAlignment;
+
+APPKIT_EXPORT const CGFloat NSGridViewSizeForContent;
+
+@class NSGridColumn, NSGridRow, NSGridCell, NSArray, NSMutableArray;
+  
+@interface NSGridView : NSView
+{
+  NSGridRowAlignment _rowAlignment;
+  NSMutableArray *_rows;
+  NSMutableArray *_columns;
+  NSMutableArray *_cells;
```

Comment:
If I were to implement these classes I would only store references to the cells here. In NSGridRow and NSGridColumn there wouldn't be any cell references and all the managing of the cells would happen here. Otherwise you have three data structures that you have to keep in sync. All the code you need to have on _cells you need anyway and for each row/column you always have access to the NSGridView to get the required information from there.

Reply:
Yep, understood.  I will take the approach you suggest here.

Reply:
Working on this now.

Reply:
This is done.  Everything is done in terms of the array in _cells.